my_variant.h

```diff
@@ -0,0 +1,118 @@
+#include <cassert>
+#include <iostream>
+#include <algorithm>
+#include <type_traits>
+
+using std::cout;
+using std::endl;
+
+
+template <typename... types>
+struct max_size_helper { };
+
+template <typename T, typename... types>
+struct max_size_helper<T, types...> {
+	static size_t constexpr value = std::max(sizeof(T), max_size_helper<types...>::value);
+};
+
+template <> 
+struct max_size_helper<> {
+	static size_t constexpr value = 0;
+};
+
+
+template <typename... types>
+struct max_align_helper { };
+
+template <typename T, typename... types>
+struct max_align_helper<T, types...> {
+	static size_t constexpr value = std::max(alignof(T), max_align_helper<types...>::value);
+};
+
+template <> 
+struct max_align_helper<> {
+	static size_t constexpr value = 0;
+};
+
+/**
+* get index of type from parameter pack
+*/
+
+template <typename T, typename... Ts> struct get_index;
+
+template <typename T, typename... Ts>
+struct get_index<T, T, Ts...> : std::integral_constant<std::size_t, 0> {};
```

Comment:
There is some inconsistency: here you use `integral_constant` (which is fine), and above helpers - manual implementation.

Reply:
Since it is an exercise, I played with 2 different styles of doing it. You are right that if it was a production code, the inconsistency is confusing.